tcms/telemetry/api/telemetry.py

```diff
@@ -0,0 +1,61 @@
+from django.db.models import Count
+from modernrpc.core import rpc_method
+
+from tcms.management.models import Priority
+from tcms.testcases.models import TestCase, Category, TestCaseStatus
+
+
+@rpc_method(name='Testing.breakdown')
+def breakdown(query=None):
+    """
+    .. function:: XML-RPC Testing.breakdown(query)
+
+        Perform a search and return the statistics for the selected test cases
+
+        :param query: Field lookups for :class:`tcms.testcases.models.TestCase`
+        :type query: dict
+        :return: Object, containing the statistics for the selected test cases
+        :rtype: dict
+    """
+
+    if query is None:
+        query = {}
+
+    test_cases = TestCase.objects.filter(**query)
+
+    count = {
+        'manual': test_cases.filter(is_automated=False).count(),
+        'automated': test_cases.filter(is_automated=True).count(),
+        'all': test_cases.count()
+    }
+
+    priorities_qs = Priority.objects.filter().annotate(
+        num_cases=Count('priority_case')
+    ).filter(
+        num_cases__gt=0,
+        priority_case__in=test_cases
+    )
+    priorities = [[priority.value, priority.num_cases] for priority in priorities_qs]
+
+    categories_qs = Category.objects.filter().annotate(
+        num_cases=Count('category_case')
+    ).filter(
+        num_cases__gt=0,
+        category_case__in=test_cases
+    )
+    categories = [[category.name, category.num_cases] for category in categories_qs]
```

Comment:
```
TestCase.objects.all().values('category').annotate(num_priority=Count('category'))
<QuerySet [{'category': 1, 'num_priority': 46}, {'category': 2, 'num_priority': 2}, {'category': 3, 'num_priority': 2}, {'category': 4, 'num_priority': 9}, {'category': 5, 'num_priority': 8}]>
```

Mind the fact that the result is a query set (aka iterator) not a dictionary. You will have to iterate over it and convert to the expected data structure for the rest of the code to continue working.